server.go

```diff
@@ -1480,7 +1485,7 @@ func (s *server) ConnectToPeer(addr *lnwire.NetAddress, perm bool) error {
 	// connect to the target peer. If the we can't make the connection, or
 	// the crypto negotiation breaks down, then return an error to the
 	// caller.
-	conn, err := brontide.Dial(s.identityPriv, addr)
+	conn, err := brontide.Dial(s.identityPriv, addr, cfg.net.Dial)
```

Comment:
could brontide be set up with the correct Dial somehow, such that you don't have to pass this everywhere? Or Maybe it would be possible to let the NetAddress resolve the correct Dialer.